runners/dmn-tck-runner-drools/src/test/java/org/omg/dmn/tck/runner/drools/DroolsTCKTest.java

```diff
@@ -530,25 +334,23 @@ private Object parseType(ValueType value, DMNType dmnType) {
                                     return LocalTime.parse(xmlCal.toXMLFormat());
                                 }
                                 return xmlCal.toGregorianCalendar();
-                            } else {
-                                return dateTimeOrDurationValue;
                             }
                         }
                     }
                 } catch (Exception e) {
-                    logger.error("Error trying to coerce JAXB type " + val.getClass().getName() + " with value '" + val.toString() + "': " + e.getMessage());
+                    LOGGER.error("Error trying to coerce JAXB type {} with value '{}': {}", val.getClass().getName(), val, e.getMessage());
```

Comment:
IINW, at line 390
`Object dateTimeOrDurationValue = (val != null) ? val.getValue() : null;`
`val` could be `null`
but here it is used for message creation, potentially throwing a NPE during error logging.

Reply:
Actually, I found out the whole try-catch block was unnecessary there. The whole code wrapped doesn't throw any checked exception. I personally don't like catching exceptions with the general Exception class, so I removed it.